src/torchcodec/decoders/_core/CudaDevice.cpp

```diff
@@ -209,8 +197,9 @@ void convertAVFrameToDecodedOutputOnCuda(
       src->format == AV_PIX_FMT_CUDA,
       "Expected format to be AV_PIX_FMT_CUDA, got " +
           std::string(av_get_pix_fmt_name((AVPixelFormat)src->format)));
-  int width = options.width.value_or(codecContext->width);
-  int height = options.height.value_or(codecContext->height);
+  int height = 0, width = 0;
+  std::tie(height, width) =
```

Comment:
IMO this is less maintainable/safe than returning a struct with named members.

Someone could write:

`std::tie(width, height)` instead of `std::tie(height, width)` and it would be hard to notice.